### ISSUE TITLE
`isModalOpen()` false negative on decorative background causes `landmark-one-main` to pass

### ISSUE THREAD

**Issue by gameroman** [dequelabs/axe-core#5258]
### Product

axe-core

### Product Version

4.12.1

### Latest Version

- [x] I have tested the issue with the latest version of the product

#### Expectation

`landmark-one-main` should report a failure on a page that has no `<main>` element, regardless of decorative background elements

#### Actual

`landmark-one-main` passes when the page contains a `position: fixed` full-viewport background element (e.g. a decorative gradient overlay), because `isModalOpen()` incorrectly classifies it as a modal, which triggers the `passForModal` short-circuit in [`has-descendant-evaluate.js`](https://github.com/dequelabs/axe-core/blob/8fcb57cd61f3483bdb8870ef85c5f584b1d8ac57/lib/checks/generic/has-descendant-evaluate.js#L11-L13)

#### How to Reproduce

```html
<!DOCTYPE html>
<html lang="en">
<head><title>Repro</title></head>
<body>
  <div style="position: fixed; inset: 0; z-index: -1; width: 100%; height: 100%; background: black; opacity: 0.4;"></div>
  <h1>Hello</h1>
  <p>No <code>main</code> element on this page.</p>
</body>
</html>
```

<https://dependents.dev>

#### Additional context

The `passForModal` option was introduced in PR #3501 to fix #3359 (real modals causing `landmark-one-main` to incorrectly fail).
That fix works correctly for actual dialogs.
However, the `isModalOpen()` heuristic behind it lacks checks that would distinguish decorative elements from real modals

**Comment by straker** [comment]
Thanks for the issue. I've confirmed the problem with the provided code. This one may take some time to solve.
